### PR TITLE
FAL-2270 Make production extra configuration configurable from config

### PR DESCRIPTION
**Pull request by samuelallan72**
[FAL-2270](https://tasks.opencraft.com/browse/FAL-2270)

This is so we don't need to update ocim code when the required configuration changes.  

The motivation is that we needed to update the config to add the sysadmin app, and thought why not make the whole config configurable instead of hardcoded.

**Test instructions**:

- deploy https://github.com/open-craft/ansible-secrets/pull/310
- launch a production instance using the prod instance factory
- verify that the sysadmin app is installed

**Reviewers**:

- [x] @pomegranited

### CODE REVIEW

File: instance/factories.py
Comment (pomegranited) [inline instance/factories.py:151]: Can you add an env variable for the version, something like `EDXAPP_EDX_SYSADMIN_VERSION`? Then we can update it when we upgrade to maple etc., without having to update the code.

**Review by pomegranited — CHANGES_REQUESTED** [review]

File: instance/factories.py
Comment (swalladge) [reply instance/factories.py:151]: @pomegranited do you mean a variable for ocim, or a variable to go in the ansible vars?

Comment (pomegranited) [reply instance/factories.py:151]: Hmm.. I was thinking a variable for Ocim, but maybe it would be better to move this whole block into config, and read it into the Ocim settings?

```yaml
PRODUCTION_INSTANCE_EXTRA_CONFIGURATION:
  # Don't create default users on production instances
  DEMO_CREATE_STAFF_USER: false
  demo_test_users: [],
  # Disable certificates process to reduce load on RabbitMQ and MySQL
  SANDBOX_ENABLE_CERTIFICATES: false
  EDXAPP_EXTRA_REQUIREMENTS
  - name: git+https://github.com/mitodl/edx-sysadmin.git@v0.1.3#egg=edx-sysadmin==v0.1.3 
```

So you can just do this:
```
production_settings = settings.PRODUCTION_INSTANCE_EXTRA_CONFIGURATION
```

That way, if/when we have other apps that we need to install on production instances, we can just add them to the configuration, including any extra env variables or whatever settings they need.

Comment (swalladge) [reply instance/factories.py:151]: Nice :ok_hand:

**Review by pomegranited — APPROVED** [review]
:+1: -- though please fix the failing unit tests before merging.

- [x] I tested this by following the https://github.com/open-craft/ansible-secrets/pull/310's instructions to test this change on ocim stage.
   Created https://stage.manage.opencraft.com/instance/8490/, which configured as expected.
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] <strike>Includes documentation</strike> N/A
- [x] <strike>I made sure any change in configuration variables is reflected in the corresponding
    client's `configuration-secure` repository.</strike> N/A

**Comment by swalladge** [comment]
Note: do not merge until [open-craft/ansible-secrets#312|https://github.com/open-craft/ansible-secrets/pull/312] is approved

**Comment by swalladge** [comment]
Merging because the failing integration test is unrelated.  See [FAL-2409](https://tasks.opencraft.com/browse/FAL-2409)

**Pull request merged**
